sound/soc/sof/xtensa/core.c

```diff
@@ -109,36 +109,19 @@ static void xtensa_stack(struct snd_sof_dev *sdev, void *oops, u32 *stack,
 {
 	struct sof_ipc_dsp_oops_xtensa *xoops = oops;
 	u32 stack_ptr = xoops->stack;
+	/* 4 dwords + 3 spaces + 1 terminating NUL */
+	unsigned char buf[8 * 4 + 3 + 1];
 	int i;
 
 	dev_err(sdev->dev, "stack dump from 0x%8.8x\n", stack_ptr);
 
-	for (i = 0; i <= stack_words - 4; i += 4) {
-		dev_err(sdev->dev, "0x%8.8x: 0x%8.8x 0x%8.8x 0x%8.8x 0x%8.8x\n",
-			stack_ptr + i, stack[i], stack[i + 1], stack[i + 2],
-			stack[i + 3]);
-	}
-
-	/* deal with any remaining words */
-	switch (stack_words - i) {
-	case 0:
-		break;
-	case 1:
-		dev_err(sdev->dev, "0x%8.8x: 0x%8.8x\n",
-			stack_ptr + stack_words - 1, stack[stack_words - 1]);
-		break;
-	case 2:
-		dev_err(sdev->dev, "0x%8.8x: 0x%8.8x 0x%8.8x\n",
-			stack_ptr + stack_words - 2, stack[stack_words - 2],
-			stack[stack_words - 1]);
-		break;
-	case 3:
-		dev_err(sdev->dev, "0x%8.8x: 0x%8.8x 0x%8.8x 0x%8.8x\n",
-			stack_ptr + stack_words - 3, stack[stack_words - 3],
-			stack[stack_words - 2], stack[stack_words - 1]);
-		break;
-	default:
-		break;
+	/* example output:
+	 * 0x0049fbb0: 8000f2d0 0049fc00 6f6c6c61 00632e63
```

Comment:
that looks different from the previous format? Does this work with the tools?

Reply:
yes, this is the result of hex_dump_to_buffer, which makes a little different format from our output. Now, for we have spent time to make it clear, we have to refine our tools according to this change.

Reply:
so if I merge this today it breaks the tools.  Not going to happen until you provide the matching change in the tools.

Reply:
@plbossart  The tools is coredumper in our tools directory. I check it and find that it has dump stack from memory windows of dsp now, not from kernel message which is also gotten from memory window (@mwierzbix is the maintainer). So this patch doesn't affect the tool now.

Reply:
ok thanks